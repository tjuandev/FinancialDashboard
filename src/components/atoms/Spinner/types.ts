import { CSSProperties } from 'react'

export type SpinnerProps = {
  size?: string
  color?: string
  bgColor?: string
  animation?: Pick<CSSProperties, 'animation'>
}
