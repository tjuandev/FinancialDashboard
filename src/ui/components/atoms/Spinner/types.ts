import { CSSProperties } from 'react'

export type SpinnerProps = {
  animation?: Pick<CSSProperties, 'animation'>
  bgColor?: string
  color?: string
  size?: string
}
