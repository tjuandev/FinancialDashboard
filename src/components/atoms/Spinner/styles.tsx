import styled from 'styled-components'
import { SpinnerProps } from './types'

export const Spinner = styled.div<SpinnerProps>`
  border-radius: 50%;
  width: 10em;
  height: 10em;

  font-size: ${({ size = 'sm' }) => size};
  position: relative;
  text-indent: -9999em;
  border: 1em solid ${({ bgColor = 'rgba(0, 0, 0, 0.2)' }) => bgColor};
  border-left: 1em solid ${({ color = '#000' }) => color};

  transform: translateZ(0);

  animation: ${({ animation = 'load 1.1s infinite linear' }) => animation};

  @keyframes load {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`
