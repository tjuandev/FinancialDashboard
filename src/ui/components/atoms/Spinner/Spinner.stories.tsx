import Spinner from '.'
import { Meta, Story } from '@storybook/react'
import { SpinnerProps } from './types'

export default {
  title: 'atoms/Spinner',
  component: Spinner
} as Meta

export const Primary: Story<SpinnerProps> = args => <Spinner {...args} />

Primary.args = {}
