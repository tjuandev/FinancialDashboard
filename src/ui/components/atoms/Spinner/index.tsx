import * as S from './styles'
import { SpinnerProps } from './types'

const Spinner = (props: SpinnerProps) => <S.Spinner {...props} />

export default Spinner
