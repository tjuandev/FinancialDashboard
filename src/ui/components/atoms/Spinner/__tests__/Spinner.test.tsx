import { render } from '@testing-library/react'

import Spinner from '..'

describe('<Spinner />', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Spinner />)

    expect(container).toMatchSnapshot()
  })
})
