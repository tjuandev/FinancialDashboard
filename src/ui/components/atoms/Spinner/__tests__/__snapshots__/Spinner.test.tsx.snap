// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> Should match snapshot 1`] = `
.c0 {
  border-radius: 50%;
  width: 10em;
  height: 10em;
  font-size: 2px;
  position: relative;
  text-indent: -9999em;
  border: 1em solid rgba(0,0,0,0.2);
  border-left: 1em solid rgba(0,0,0,0.7);
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: load 0.8s infinite linear;
  animation: load 0.8s infinite linear;
}

<div>
  <div
    class="c0"
  />
</div>
`
